src/Catalyst.Abstractions/Kvm/Models/TransactionForRpc.cs

```diff
@@ -21,74 +21,60 @@
 
 #endregion
 
-using System.Numerics;
+using LibP2P;
 using Nethermind.Core;
 using Nethermind.Core.Crypto;
-using Nethermind.Core.Extensions;
+using Nethermind.Core.Json;
 using Nethermind.Dirichlet.Numerics;
+using Newtonsoft.Json;
 
 namespace Catalyst.Abstractions.Kvm.Models
 {
     public class TransactionForRpc
     {
-        public TransactionForRpc(Keccak blockHash, BigInteger? blockNumber, int? txIndex, Transaction transaction)
-        {
-            Hash = transaction.Hash;
-            Nonce = transaction.Nonce;
-            BlockHash = blockHash;
-            BlockNumber = blockNumber;
-            TransactionIndex = txIndex;
-            From = transaction.SenderAddress;
-            To = transaction.To;
-            Value = transaction.Value;
-            GasPrice = transaction.GasPrice;
-            Gas = transaction.GasLimit;
-            Input = Data = transaction.Data ?? transaction.Init;
-            R = transaction.Signature?.R;
-            S = transaction.Signature?.S;
-            V = (UInt256?) transaction.Signature?.V;
-        }
-
-        // ReSharper disable once UnusedMember.Global
-        public TransactionForRpc() { }
-
+        [JsonConverter(typeof(KeccakConverter))]
         public Keccak Hash { get; set; }
```

Comment:
Should this be Blake2b? Is there any reason why we are using Keccak? Compatibility or just a placeholder?